lib/matplotlib/patches.py

```diff
@@ -4182,49 +4182,90 @@ def __init__(self, posA=None, posB=None, *,
-          clipped with *patchA* and *patchB* and further shrunken by
-          *shrinkA* and *shrinkB*. An arrow is drawn along this
-          resulting path using the *arrowstyle* parameter.
+        There are two ways to define the arrow position and path:
 
-        - Alternatively if *path* is provided, an arrow is drawn along this
-          path and *patchA*, *patchB*, *shrinkA*, and *shrinkB* are ignored.
+        - **Start, end and connection**:
+          The typical approach is to define the start and end points of the
+          arrow using *posA* and *posB*. The curve between these two can
+          further be configured using *connectionstyle*.
+
+          If given, the arrow curve is clipped by *patchA* and *patchB*,
+          allowing it to start/end at the border of these patches.
+          Additionally, the arrow curve can be shortened by *shrinkA* and
+          to create a margin between the arrow and the patch it is pointing
+          to.
+
+        - **path**: Alternatively if *path* is provided, an arrow is drawn along
+          this Path. In this case, *connectionstyle*, *patchA*, *patchB*,
+          *shrinkA*, and *shrinkB* are ignored.
+
+        **Styling**
+
+        The *arrowstyle* defines the styling of the arrow head, tail and shaft.
+        The resulting arrows can be styled further by setting the `.Patch`
+        properties such as *linewidth*, *color*, *facecolor*, *edgecolor*
+        etc. via keyword arguments.
 
         Parameters
         ----------
-        posA, posB : (float, float), default: None
-            (x, y) coordinates of arrow tail and arrow head respectively.
+        posA, posB : (float, float), optional
+            (x, y) coordinates of start and end point of the arrow.
+            The actually drawn start and end positions may be modified
+            through *patchA*, *patchB*, *shrinkA*, and *shrinkB*.
 
-        path : `~matplotlib.path.Path`, default: None
+            *posA*, *posB* are exclusive of *path*.
+
+        path : `~matplotlib.path.Path`, optional
             If provided, an arrow is drawn along this path and *patchA*,
             *patchB*, *shrinkA*, and *shrinkB* are ignored.
 
+            *path* is exclusive of *posA*, *posB*.
+
         arrowstyle : str or `.ArrowStyle`, default: 'simple'
-            The `.ArrowStyle` with which the fancy arrow is drawn.  If a
-            string, it should be one of the available arrowstyle names, with
-            optional comma-separated attributes.  The optional attributes are
-            meant to be scaled with the *mutation_scale*.  The following arrow
-            styles are available:
+            The styling of arrow head, tail and shaft. This can be
+
+            - `.ArrowStyle` or one of its subclasses
+            - The shorthand string name as given in the table below, e.g. "->",
+              optionally containing a comma-separated list of style parameters,
```

Comment:
The parenthetical there somewhat conflicts with the following clause.
```suggestion
            - The shorthand string name (e.g. "->") as given in the table below,
              optionally containing a comma-separated list of style parameters,
```